app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -86,6 +86,25 @@ import { addSearchSuggestion } from "discourse/widgets/search-menu-results";
 // If you add any methods to the API ensure you bump up this number
 const PLUGIN_API_VERSION = "0.12.2";
 
+// This helper prevents us from applying the same `modifyClass` over and over in test mode.
+function canModify(klass, type, resolverName, changes) {
+  if (!changes.pluginId) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      "To prevent errors, add a `pluginId` key to your changes when calling `modifyClass`"
+    );
+    return true;
+  }
+
+  let key = "_" + type + "/" + changes.pluginId + "/" + resolverName;
+  if (klass.class[key]) {
```

Comment:
`class` is a bit cryptic name for this property, but I can't say I have a better suggestion 😉

Reply:
That's actually the "target" from before: `klass.class.reopenClass` - an ember internal I believe. The only thing I made up here is the `key`